pages/404.js

```diff
@@ -1,6 +1,16 @@
 import Link from 'next/link'
+import { useSession } from 'next-auth/react'
 
 export default function Custom404() {
+  const { status } = useSession({
+    required: true,
+    onUnauthenticated() {
+      // The user is not authenticated, handle it here.
+      console.log('User is not logged in')
+    },
+  })
+  console.log('User Status: ', status)
```

Comment:
Can this use the existing logic within my-dashboard to check if users are logged in?
Check ./lib/auth AuthIsValid() to check if users are authenticated